pkg/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -2,10 +2,13 @@ package deployment
 
 import (
 	edgeconnectv1alpha1 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1alpha1/edgeconnect"
-	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/edgeconnect/consts"
-	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects"
+	edgeconnectconsts "github.com/Dynatrace/dynatrace-operator/pkg/controllers/edgeconnect/consts"
 	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/address"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/resources"
+	utilmap "github.com/Dynatrace/dynatrace-operator/pkg/util/map"
 	"github.com/Dynatrace/dynatrace-operator/pkg/util/prioritymap"
+	utilconsts "github.com/Dynatrace/dynatrace-operator/pkg/util/testing/consts"
```

Comment:
this is the reason that I really dislike that consts package, you are using a "testing" package in live code.

that consts package should be destroyed and its contents moved to where they are used, if there is duplication that is fine, as those will be in something_test.go

if there are contents of the package that are actually not test related, and could be reused then we can find a place for them, for example in pkg/consts (not a fan of the package, but still better for stuff like UnprivilegedGroup)

Reply:
I managed to split up consts so far that only some constants are left with the prefix "Test". I could as well remove those too and define for example TestAppName multiple times everytime where it is used. But I think it is better to store it into pkg/consts and leave it there into a file called test_constants.go. So basically only some very general constants are left in there. What do you think?
done in https://github.com/Dynatrace/dynatrace-operator/pull/2336/commits/1b08b6e411e27410ac3bd3eaa85ff6d96ad1cb8f

Reply:
why do we want to have tests related consts in `pkg/consts` directory? `test/consts` seems to be better place

Reply:
pls do not make tests in different packages reliant of a central test consts, that is asking for trouble, I personally even avoid using (package) globally defined test consts, and define them on the function level, even if I duplicate stuff. So each test case is on its own.
```
func TestFindVersionFromFileSystem(t *testing.T) {
	testPath := "/test"
	testVersion := "1.239.14.20220325-164521"
	t.Run("get version from directory in file system", func(t *testing.T) {...}
```

if I want to create a more "complex" struct, then I put it into a function like `func createTestDeployment` or something.

I would only see value in a shared testing package IF it contained more complicated stuff then consts and some testing structs, example: https://github.com/Dynatrace/dynatrace-operator/blob/d5f09dbd926829de49766ba1f502f34050fd0803/pkg/util/testing/events.go#L20-L20

Reply:
my original comment was trying to say:
- move NOT testing related consts somewhere else, maybe `pkg/consts`
- move TESTING related consts into the package where they are used, if you need to duplicate that is fine
   - you are trying to protect the unit tests of the (old) `kubeobjects` package, but those tests are just not good when it comes to best practices, they share things in a very weird a volatile way, I don't want to preserve this "sharing" of code. So I would chose a "lesser" problem by just duplicating what was shared, so incase 1 package needs some extra love, then it will be easier to fix. (so people will not just update the shared consts to fit their need and then get surprised that other package unit tests are failing)

Reply:
Okay I get your arguments. I will do that. done in https://github.com/Dynatrace/dynatrace-operator/pull/2336/commits/4a344b515daf49c653a7d0c56748d524fc10421f